jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ProjectProperties.java

```diff
@@ -46,21 +48,38 @@ String getMainClass(@Nullable String mainClass) {
     if (mainClass == null) {
       mainClass = getMainClassFromJarTask();
       if (mainClass == null) {
-        throw new GradleException(
-            HelpfulSuggestionsProvider.get("Could not find main class specified in a 'jar' task")
-                .suggest("add a `mainClass` configuration to jib"));
+        gradleBuildLogger.info(
+            "Could not find main class specified in a 'jar' task; attempting to "
+                + "infer main class.");
+        try {
+          mainClass =
+              MainClassFinder.findMainClass(
+                  project
+                      .getBuildDir()
+                      .toPath()
+                      .resolve("classes")
+                      .resolve("java")
+                      .resolve("main")
+                      .toAbsolutePath()
+                      .toString());
+        } catch (MultipleClassesFoundException | IOException ex) {
+          throw new GradleException(
+              HelpfulSuggestionsProvider.get("Failed to get main class: " + ex.getMessage())
+                  .suggest("add a `mainClass` configuration to jib"));
```

Comment:
The `ex` should go as a second argument to `GradleException` rather than a string addition so there is stack hierarchy.